app/models/authenticator.rb

```diff
@@ -1,24 +1,14 @@
 class Authenticator < Struct.new(:auth_hash, :session)
-  attr_reader :user
-
   def perform
-    find_or_create_user
-    update_user
+    user.from_oauth_hash(auth_hash)
     sign_in
     check_sam
-    redirect_hash
   end
 
   private
 
-  def find_or_create_user
-    @user = found_user || created_user
-  end
-
-  def update_user
-    if user.name.blank?
-      user.update(name: name)
-    end
+  def user
+    @_user ||= User.find_or_create_by(github_id: auth_hash[:uid])
   end
 
   def sign_in
```

Comment:
I myself am not too worried about it... maybe we should open a GH issue for this?